pkg/sloop/webserver/debug.go

```diff
@@ -162,6 +164,117 @@ func listKeysHandler(tables typed.Tables) http.HandlerFunc {
 	}
 }
 
+type sloopKeyInfo struct {
+	MinimumSize int64
+	MaximumSize int64
+	TotalKeys   int64
+	TotalSize   int64
+	AverageSize float64
+}
+
+type sloopKey struct {
+	TableName   string
+	PartitionID string
+}
+
+type histogram struct {
+	HistogramMap map[sloopKey]*sloopKeyInfo
+	TotalKeys    int
+	DeletedKeys  int
+}
+
+// returns IsDeletedExpired, TableName, PartitionId, TotalSize, error.
+func parseSloopKey(item badgerwrap.Item) (bool, string, string, int64, error) {
```

Comment:
- I'm not seeing much value in returning `item.IsDeletedOrExpired()` or `item.EstimatedSize()` as part of this function instead of forcing the caller to get it
- by removing them, the method name also fits better
- it might be convenient for this method to return a `sloopKey`

Reply:
https://github.com/salesforce/sloop/blob/5cd57b189c75a3e801b98d64a47aa9fe159fa576/pkg/sloop/webserver/debug.go#L230 isDeletedOrExpired is used here. Similarly size is also used.

Reply:
yes - but they are not used inside this method, they don't really have anything to do with parsing the key, and the caller can just as easily call the accessors to get the values.